pkg/platform/runtime/store/store.go

```diff
@@ -70,22 +72,41 @@ func (s *Store) HasMarker() bool {
 	return false
 }
 
-// MatchesCommit checks if stored runtime is complete and can be loaded
-func (s *Store) MatchesCommit(commitID strfmt.UUID) bool {
+// MarkerIsValid checks if stored runtime is complete and can be loaded
+func (s *Store) MarkerIsValid(commitID strfmt.UUID) bool {
 	marker := s.markerFile()
 	if !fileutils.FileExists(marker) {
 		logging.Debug("Marker does not exist: %s", marker)
 		return false
 	}
 
-	contents, err := fileutils.ReadFile(marker)
+	f, err := os.OpenFile(marker, os.O_RDONLY, 0)
 	if err != nil {
-		logging.Error("Could not read marker: %v", err)
+		logging.Error("Could not open marker file: %v", err)
 		return false
 	}
+	defer f.Close()
 
-	logging.Debug("MatchesCommit for %s, %s==%s", marker, string(contents), commitID.String())
-	return strings.TrimSpace(string(contents)) == commitID.String()
+	sc := bufio.NewScanner(f)
+	ok := sc.Scan()
+	if !ok {
+		logging.Error("Expected commit ID in marker file: %v", sc.Err())
+		return false
+	}
+	parsedID := strings.TrimSpace(sc.Text())
+	logging.Debug("Matching commitID for %s, %s==%s", marker, parsedID, commitID.String())
+	if parsedID != commitID.String() {
```

Comment:
Might be good to debug log the expected vs actual here (same for state tool version further down).